src/analysis/normed_space/weak_dual.lean

```diff
@@ -30,6 +32,8 @@ The main definitions concern the canonical mapping `dual 𝕜 E → weak_dual 
 * `normed_space.dual.continuous_linear_map_to_weak_dual`: A continuous linear mapping from
   `dual 𝕜 E` to `weak_dual 𝕜 E` (same as `normed_space.dual.to_weak_dual` but different bundled
   data).
+* `polar s` is the subset of `weak_dual 𝕜 E` consisting of those functionals `x'` for which
```

Comment:
I am wondering why you define it as a subset of `weak_dual` and not of `dual`. Most properties in this file do not depend on the weak topology (as far as I can tell, only `is_closed` does). So, I'd move everything to the file `dual`, and only add here the fact that `polar 𝕜 s`, seen as a subset of the weak dual, is closed.

Reply:
Fair point --- my reason for this choice was that I use weak star topology for polar sets in Banach-Alaoglu, and this is clearly not a good reason to define them in the `weak_dual`.

I changed this. Almost everything else is unaffected, as you said, but for the proof of `polar.is_closed` I needed two new lemmas: `polar.to_weak_dual_image_norm_eval_le_one` and `polar.of_empty`.

Reply:
Ah, I changed the definition, but did not yet move this to the file `dual`. I will now.

Reply:
In the file-level docstring in `dual.lean` I now added a "main definitions" subsection, and I tried to do justice to the existing definitions while I also added the mention of the definition of `polar`. I did not yet dare to add "main results" there.

Moving most things about polar sets to `dual.lean` also required two additional imports in `dual.lean`.

These changes may not be uncontroversial. So let us also ask @hrmacbeth.

Reply:
It's fine with me, although I think the `seminorm` import is unnecessary!